### PR TITLE
Feat bt colormap

### PR DESCRIPTION
**Pull request by tdivoll**
This PR is for a colormap legend to go with the chlorophyll markers on the map view.

### CODE REVIEW

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 1]
```diff
@@ -120,9 +122,16 @@ const props = defineProps<{
 
 const config = {
   chlorophyll: {
-    varDomain: [0, 10],
-    markerSize: [0.2, 0.75],
-    markerColors: ["#32CD32", "#355E3B"], // lime green, hunter green
+    varDomain: [0.1, 1, 10, 100, 1000, 10000],
+    markerSize: [0.1, 0.5],
```
Comment (tdivoll) [inline src/components/buoy-telemetry/BuoyMap.vue:126]: Leaving the max marker size at 1 made it really large even at small values

**Comment by tdivoll** [comment]
It works locally, but the version in the PR link is not displaying. I will debug. 
image

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 2]
```diff
@@ -353,4 +391,34 @@ watch(() => props.formattedDate, updateMap);
...
+  padding: 10px;
+  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.75);
+  line-height: 28px;
+  height: 250px;
+  margin-bottom: 10px;
+  width: 115px;
+}
+
+/* prettier-ignore */
+:deep(.legend-key){
```
Comment (eldu) [inline src/components/buoy-telemetry/BuoyMap.vue:417]: Oooh so this is the magic sauce! 🥳

**Review by eldu — APPROVED** [review]
This looks great! 

The package-lock.json has some changes, was it just an update? I would have expected some changes in the package.json as well. No worries, just asking!

File: src/components/buoy-telemetry/BuoyMap.vue [hunk 3]
```diff
@@ -146,7 +155,7 @@ const annotatedChlorophyllSamples = computed(() => {
         return {
           ...row,
           color: colorScale(row.value),
-          size: sqrtScale(row.value),
+          size: logScale(row.value),
```
Comment (mirestrepo) [inline src/components/buoy-telemetry/BuoyMap.vue:158]: Do we need to worry about an undefined for log(0)?

**Review by mirestrepo — APPROVED** [review]
Looks great. Just a question about log(0)

**Comment by tdivoll** [comment]
> This looks great!
> 
> The package-lock.json has some changes, was it just an update? I would have expected some changes in the package.json as well. No worries, just asking!

I think some things got updated when I installed `d3-svg-legend`, and then not everything downgraded back to what it was once I removed `d3-svg-legend`. It is on my list to walk through the dependencies and see what we can upgrade without breaking changes.

File: src/components/buoy-telemetry/BuoyMap.vue (hunk 3 above)
Comment (tdivoll) [reply src/components/buoy-telemetry/BuoyMap.vue:158]: It seems to work okay when the chlorophyll value is zero. That is the case in the current data and it just doesn't display anything on that buoy.

Comment (mirestrepo) [reply src/components/buoy-telemetry/BuoyMap.vue:158]: Sounds good. Maybe add a comment on that line about it, in case it ever cases an issue

**Pull request merged**